src/main/kotlin/lib/factories/SparkMaxFactory.kt

```diff
@@ -0,0 +1,119 @@
+     *  idle mode: brake
+     *  inverted: false
+     *  current limit: 30 amps
+     * @return a spark max configured to the default settings
+     */
+    data class SparkWithConfig(val spark: CANSparkMax, val config: SparkMaxConfig = SparkMaxConfig())
+
+    companion object {
+        const val MAX_CAN_FRAME_PERIOD = 65535
+
+        // A list of all configured sparks linked to thier configs
+        val listOfAllSparksAndConfigs: MutableList<SparkWithConfig> = ArrayList()
+        fun updateCanFramePeriods() {
+            for (sparkAndConfig in listOfAllSparksAndConfigs) {
+                val spark = sparkAndConfig.spark
+                val config = sparkAndConfig.config
+                if (spark.getStickyFault(CANSparkMax.FaultID.kHasReset)) {
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus0, config.frame0Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus1, config.frame1Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus2, config.frame2Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus3, config.frame3Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus4, config.frame4Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus5, config.frame5Rate) }
+                    RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus6, config.frame6Rate) }
+                }
+            }
+        }
+
+        fun rerunConfigs() {
+            for (sparkAndConfig in listOfAllSparksAndConfigs) {
+                val spark = sparkAndConfig.spark
+                val config = sparkAndConfig.config
+
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus0, config.frame0Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus1, config.frame1Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus2, config.frame2Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus3, config.frame3Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus4, config.frame4Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus5, config.frame5Rate) }
+                RevUtil.autoRetry { spark.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus6, config.frame6Rate) }
+
+                RevUtil.autoRetry { spark.setIdleMode(config.idleMode) }
+                spark.inverted = config.inverted
```

Comment:
This behavior is different than the one in `createSparkMax`, which deals with following. This one dosen't deal with following at all